crates/shared/src/ethrpc/buffered.rs

```diff
@@ -43,7 +43,7 @@ impl Default for Configuration {
 /// Buffered `Transport` implementation that implements automatic batching of
 /// JSONRPC requests.
 #[derive(Clone, Debug)]
-pub struct Buffered<Inner> {
+pub struct BufferedTransport<Inner> {
```

Comment:
Although I do think `BatchingTransport` might be a slightly better name. `BatchTransport` is a name already used by the `web3` crate to mean something different (a transport allowing batches), but I don't think that's a problem.

Reply:
I called it `Buffered` instead of `Batched` just because it does primarily request buffering (and, as a side-effect, automatically groups buffered requests into batches).

For example:
```rust
Configuration {
    max_batch_size: 1,
    max_concurrent_requests: 5,
    ..Default::default()
}
```

Will just buffer requests and make sure only 5 at a time are sent to the node, but never make any batched RPC requests.

Don't know if this makes sense. Also fine with changing it, I'm not too attached to the name (and "buffered" is quite generic lacking specificity).